xclim/indices/_agro.py

```diff
@@ -589,6 +595,225 @@ def water_budget(
+        [0, 1],
+        dims=["dparams"],
+        coords=dict(dparams=(["loc", "scale"])),
+        attrs=dict(scipy_dist="norm"),
+    )
+    spi = dist_method("ppf", params_norm, prob)
+    spi.attrs["units"] = ""
+    spi.attrs["standard name"] = "SPI"
+    spi.attrs["long name"] = "Standardized Precipitation Index"
+
+    return spi
+
+
+@declare_units(
+    wb="[precipitation]",
+    wb_cal="[precipitation]",
+)
+def standardized_precipitation_evapotranspiration_index(
+    wb: xarray.DataArray,
+    wb_cal: xarray.DataArray,
+    freq: str = "MS",
+    window: int = 1,
+    dist: str = "gamma",
+    method: str = "ML",
+) -> xarray.DataArray:
+    r"""Standardized Precipitation Evapotranspiration Index (SPEI).
+
+    Precipitation minus potential evapotranspiration data (PET) fitted to a statistical distribution (dist), transformed to a cdf,  and inverted back to a gaussian normal pdf. The potential evapotranspiration is calculated with a given method (method).
+
+    Parameters
+    ----------
+    wb : xarray.DataArray
+      Daily water budget (pr - pet).
+    wb_cal : xarray.DataArray
+      Daily water budget used for calibration.
+    freq : str
+      Resampling frequency. A monthly or daily frequency is expected.
+    window : int
+      Averaging window length relative to the resampling frequency. For example, if `freq="MS"`, i.e. a monthly resampling, the window
+      is an integer number of months.
+    dist : str
+      Name of the univariate distribution, such as `beta`, `expon`, `genextreme`, `gamma`, `gumbel_r`, `lognorm`, `norm`
+      (see :py:mod:`scipy.stats`).
+    method : str
+      Name of the fitting method, such as `ML` (maximum likelihood), `APP` (approximate). The approximate method
+      simply uses the parameters output of `_fit_start`.
+
+    Returns
+    -------
+    xarray.DataArray,
+      Standardized Precipitation Evapotranspiration Index.
+
+    Notes
+    -----
+    See Standardized Precipitation Index (SPI) for more details on usage.
+    """
+    # Allowed distributions are constrained by the SPI function
+    if dist in ["gamma", "pearson3"]:
+        # Distributions bounded by zero: Water budget must be shifted, only positive values
+        # are allowed. The offset choice is arbitrary and needs to be revisited.
+        # In monocongo, the offset would be 1000/(60*60*24) in [kg m-2 s-1]
+        # The choice can lead to differences as big as +/-0.2 in the SPEI.
+        # If taken too big, there are problems with the "ML" method  (this should be an
+        # issue with the fitting procedure that also needs attention)
+        offset = xarray.DataArray([1e-04])
+        offset.attrs["units"] = "kg m-2 s-1"
+        convert_units_to(offset, wb.units)
+        offset = offset.values.item()
+        # Increase offset if negative values remain
+        if wb.min() + offset < 0 or wb_cal.min() + offset < 0:
+            offset = offset - 2 * min(wb.min(), wb_cal.min())
```

Comment:
You can't test a DataArray within a `if` because it might be using dask, which means it hasn't been computed yet and the user doesn't expect a computation right now.

Is this related to our previous discussion? I mean, can't you simply always  add the full offset? It could then look like:
```suggestion
         offset = offset - 2 * min(wb.min(), wb_cal.min(), 0)
```